bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/AuditorCheckAllLedgersTask.java

```diff
@@ -0,0 +1,286 @@
+                               ShutdownTaskHandler shutdownTaskHandler)
+            throws UnavailableException {
+        super(conf, auditorStats, admin, ledgerManager,
+                ledgerUnderreplicationManager, shutdownTaskHandler);
+
+        if (conf.getAuditorMaxNumberOfConcurrentOpenLedgerOperations() <= 0) {
+            LOG.error("auditorMaxNumberOfConcurrentOpenLedgerOperations should be greater than 0");
+            throw new UnavailableException("auditorMaxNumberOfConcurrentOpenLedgerOperations should be greater than 0");
+        }
+        this.openLedgerNoRecoverySemaphore =
+                new Semaphore(conf.getAuditorMaxNumberOfConcurrentOpenLedgerOperations());
+
+        if (conf.getAuditorAcquireConcurrentOpenLedgerOperationsTimeoutMSec() < 0) {
+            LOG.error("auditorAcquireConcurrentOpenLedgerOperationsTimeoutMSec should be greater than or equal to 0");
+            throw new UnavailableException("auditorAcquireConcurrentOpenLedgerOperationsTimeoutMSec "
+                    + "should be greater than or equal to 0");
+        }
+        this.openLedgerNoRecoverySemaphoreWaitTimeoutMSec =
+                conf.getAuditorAcquireConcurrentOpenLedgerOperationsTimeoutMSec();
+
+        this.ledgerCheckerExecutor = Executors.newSingleThreadExecutor(new ThreadFactory() {
+            @Override
+            public Thread newThread(Runnable r) {
+                Thread t = new Thread(r, "AuditorCheckAllLedgers-LedgerChecker");
+                t.setDaemon(true);
+                return t;
+            }
+        });
+    }
+
+    @Override
+    protected void runTask() {
+        Stopwatch stopwatch = Stopwatch.createStarted();
+        boolean checkSuccess = false;
+        try {
+            if (!isLedgerReplicationEnabled()) {
+                LOG.info("Ledger replication disabled, skipping checkAllLedgers");
+                return;
+            }
+
+            LOG.info("Starting checkAllLedgers");
+            checkAllLedgers();
+            long checkAllLedgersDuration = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS);
+            LOG.info("Completed checkAllLedgers in {} milliSeconds", checkAllLedgersDuration);
+            auditorStats.getCheckAllLedgersTime()
+                    .registerSuccessfulEvent(checkAllLedgersDuration, TimeUnit.MILLISECONDS);
+            checkSuccess = true;
+        } catch (InterruptedException ie) {
+            Thread.currentThread().interrupt();
+            LOG.error("Interrupted while running periodic check", ie);
+        } catch (BKException bke) {
+            LOG.error("Exception running periodic check", bke);
+        } catch (IOException ioe) {
+            LOG.error("I/O exception running periodic check", ioe);
+        } catch (ReplicationException.NonRecoverableReplicationException nre) {
+            LOG.error("Non Recoverable Exception while reading from ZK", nre);
+            submitShutdownTask();
+            submitShutdownTask();
```

Comment:
remove the duplicated `submitShutdownTask()`?

Reply:
Good catch. I have fixed it.